<a href="https://colab.research.google.com/github/Marzkartella/Tella1/blob/main/212_Train_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder


In [2]:
# Load dataset
train_dataset = pd.read_csv('/content/train.csv')

# Display the first few rows of the dataset
print(train_dataset.head())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [3]:
train_dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [4]:
train_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [5]:
train_dataset.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

**Passenger ID**

  Count: 891

  Mean: 446.0

  Min - Max: 1 - 891

  Unique identifier for each passenger; no missing values.


**Survived**

  Count: 891

  Mean: 0.384 (38.4% of passengers survived)

  Min - Max: 0 - 1

  Binary variable: 0 (Did not survive), 1 (Survived)

  More than half of the passengers did not survive, indicating that survival was limited to certain groups.


**Passenger Class (Pclass)**

  Count: 891

  Mean: 2.31

  Std Dev: 0.836

  Min - Max: 1 - 3

  The majority of passengers traveled in 3rd class, which may have had a higher mortality rate due to limited access to safety measures.


**Age**

  Count: 714 (177 missing values)

  Mean: 29.70 years

  Std Dev: 14.53

  Min - Max: 0.42 - 80

  Younger passengers, particularly children, may have had higher survival rates due to priority evacuation.


**Number of Siblings/Spouses Aboard (SibSp)**

  Count: 891

  Mean: 0.523

  Std Dev: 1.103

  Min - Max: 0 - 8

  Passengers traveling with family might have had better survival chances due to group coordination.


**Number of Parents/Children (Parch)**

  Count: 891

  Mean: 0.382

  Std Dev: 0.806

  Min - Max: 0 - 6

  Families with children may have had priority access to lifeboats.


**Fare**

  Count: 891

  Mean: 32.20

  Std Dev: 49.69

  Min - Max: 0 - 512.33

  Higher fares were associated with first-class passengers, who likely had a higher survival rate.

In [6]:
train_dataset.shape

(891, 12)

In [7]:
train_dataset.isna()

PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  Ticket  \
0          False     False   False  False  False  False  False  False   False   
1          False     False   False  False  False  False  False  False   False   
2          False     False   False  False  False  False  False  False   False   
3          False     False   False  False  False  False  False  False   False   
4          False     False   False  False  False  False  False  False   False   
..           ...       ...     ...    ...    ...    ...    ...    ...     ...   
886        False     False   False  False  False  False  False  False   False   
887        False     False   False  False  False  False  False  False   False   
888        False     False   False  False  False   True  False  False   False   
889        False     False   False  False  False  False  False  False   False   
890        False     False   False  False  False  False  False  False   False   

      Fare  Cabin  Embarked  
0    False   True     False  
1    False  False     False  
2    False   True     False  
3    False  False     False  
4    False   True     False  
..     ...    ...       ...  
886  False   True     False  
887  False  False     False  
888  False   True     False  
889  False  False     False  
890  False   True     False  

[891 rows x 12 columns]

In [8]:
train_dataset_update = train_dataset.drop(columns=['Cabin'])
print(train_dataset)

     PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ... 

In [9]:
train_dataset_update.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Embarked  
0      0         A/5 21171   7.2500        S  
1      0          PC 17599  71.2833        C  
2      0  STON/O2. 3101282   7.9250        S  
3      0            113803  53.1000        S  
4      0            373450   8.0500        S

In [10]:
train_dataset.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [11]:
print(train_dataset["Age"].max())

80.0


In [12]:
survival_counts = train_dataset['Survived'].value_counts()
print(survival_counts)

Survived
0    549
1    342
Name: count, dtype: int64


In [13]:
cabins_counts = train_dataset['Cabin'].value_counts()
print(cabins_counts)

Cabin
G6             4
C23 C25 C27    4
B96 B98        4
F2             3
D              3
              ..
E17            1
A24            1
C50            1
B42            1
C148           1
Name: count, Length: 147, dtype: int64


In [14]:
# Initialize LabelEncoder
label_encoder = LabelEncoder()

# Apply Label Encoding to Sex categorical column
train_dataset_update['Sex'] = label_encoder.fit_transform(train_dataset_update['Sex'])

In [15]:
print(train_dataset_update)

     PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name  Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    1  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1   
2                               Heikkinen, Miss. Laina    0  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1   
4                             Allen, Mr. William Henry    1  35.0      0   
..                                                 ...  ...   ...    ...   
886        

In [16]:
train_dataset_update.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    1  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1      0   
2                             Heikkinen, Miss. Laina    0  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1      0   
4                           Allen, Mr. William Henry    1  35.0      0      0   

             Ticket     Fare Embarked  
0         A/5 21171   7.2500        S  
1          PC 17599  71.2833        C  
2  STON/O2. 3101282   7.9250        S  
3            113803  53.1000        S  
4            373450   8.0500        S

In [17]:
train_dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [18]:
train_dataset_update1 = train_dataset.drop(columns=['Name', 'SibSp', 'Parch', 'Ticket', 'Cabin'])
print(train_dataset_update1)

     PassengerId  Survived  Pclass     Sex   Age     Fare Embarked
0              1         0       3    male  22.0   7.2500        S
1              2         1       1  female  38.0  71.2833        C
2              3         1       3  female  26.0   7.9250        S
3              4         1       1  female  35.0  53.1000        S
4              5         0       3    male  35.0   8.0500        S
..           ...       ...     ...     ...   ...      ...      ...
886          887         0       2    male  27.0  13.0000        S
887          888         1       1  female  19.0  30.0000        S
888          889         0       3  female   NaN  23.4500        S
889          890         1       1    male  26.0  30.0000        C
890          891         0       3    male  32.0   7.7500        Q

[891 rows x 7 columns]


In [19]:
train_dataset_update1.head()

PassengerId  Survived  Pclass     Sex   Age     Fare Embarked
0            1         0       3    male  22.0   7.2500        S
1            2         1       1  female  38.0  71.2833        C
2            3         1       3  female  26.0   7.9250        S
3            4         1       1  female  35.0  53.1000        S
4            5         0       3    male  35.0   8.0500        S

In [20]:
# Fill missing values in 'Age' with the median of 'Age' column
train_dataset_update1['Age'].fillna(train_dataset_update1['Age'].median(), inplace=True)

print(train_dataset_update1)

     PassengerId  Survived  Pclass     Sex   Age     Fare Embarked
0              1         0       3    male  22.0   7.2500        S
1              2         1       1  female  38.0  71.2833        C
2              3         1       3  female  26.0   7.9250        S
3              4         1       1  female  35.0  53.1000        S
4              5         0       3    male  35.0   8.0500        S
..           ...       ...     ...     ...   ...      ...      ...
886          887         0       2    male  27.0  13.0000        S
887          888         1       1  female  19.0  30.0000        S
888          889         0       3  female  28.0  23.4500        S
889          890         1       1    male  26.0  30.0000        C
890          891         0       3    male  32.0   7.7500        Q

[891 rows x 7 columns]


<ipython-input-20-0493c8821614>:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  train_dataset_update1['Age'].fillna(train_dataset_update1['Age'].median(), inplace=True)


In [21]:
train_dataset_update1.head(891)

PassengerId  Survived  Pclass     Sex   Age     Fare Embarked
0              1         0       3    male  22.0   7.2500        S
1              2         1       1  female  38.0  71.2833        C
2              3         1       3  female  26.0   7.9250        S
3              4         1       1  female  35.0  53.1000        S
4              5         0       3    male  35.0   8.0500        S
..           ...       ...     ...     ...   ...      ...      ...
886          887         0       2    male  27.0  13.0000        S
887          888         1       1  female  19.0  30.0000        S
888          889         0       3  female  28.0  23.4500        S
889          890         1       1    male  26.0  30.0000        C
890          891         0       3    male  32.0   7.7500        Q

[891 rows x 7 columns]

In [22]:
# Calculate the mode for the 'Embarked' column
mode_values = train_dataset_update1['Embarked'].mode()[0]

# Fill missing values in 'Embarked' with the mode
train_dataset_update1['Embarked'].fillna(mode_values, inplace=True)

print(train_dataset_update1)

     PassengerId  Survived  Pclass     Sex   Age     Fare Embarked
0              1         0       3    male  22.0   7.2500        S
1              2         1       1  female  38.0  71.2833        C
2              3         1       3  female  26.0   7.9250        S
3              4         1       1  female  35.0  53.1000        S
4              5         0       3    male  35.0   8.0500        S
..           ...       ...     ...     ...   ...      ...      ...
886          887         0       2    male  27.0  13.0000        S
887          888         1       1  female  19.0  30.0000        S
888          889         0       3  female  28.0  23.4500        S
889          890         1       1    male  26.0  30.0000        C
890          891         0       3    male  32.0   7.7500        Q

[891 rows x 7 columns]


<ipython-input-22-54582932a372>:5: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  train_dataset_update1['Embarked'].fillna(mode_values, inplace=True)


In [23]:
train_dataset_update1.head(891)

PassengerId  Survived  Pclass     Sex   Age     Fare Embarked
0              1         0       3    male  22.0   7.2500        S
1              2         1       1  female  38.0  71.2833        C
2              3         1       3  female  26.0   7.9250        S
3              4         1       1  female  35.0  53.1000        S
4              5         0       3    male  35.0   8.0500        S
..           ...       ...     ...     ...   ...      ...      ...
886          887         0       2    male  27.0  13.0000        S
887          888         1       1  female  19.0  30.0000        S
888          889         0       3  female  28.0  23.4500        S
889          890         1       1    male  26.0  30.0000        C
890          891         0       3    male  32.0   7.7500        Q

[891 rows x 7 columns]

In [24]:
import pandas as pd

# Load the dataset
train_dataset_update1 = pd.read_csv('train.csv')

# Count the number of survivors and non-survivors
survival_counts = train_dataset_update1['Survived'].value_counts()

print(survival_counts)

Survived
0    549
1    342
Name: count, dtype: int64


In [25]:
# Calculate the median of the Age column
median_age = train_dataset_update1['Age'].median()

print("Median Age:", median_age)

Median Age: 28.0


In [26]:
# Group by 'Pclass' and calculate the median age for each class
median_age_by_class = train_dataset_update1.groupby('Pclass')['Age'].median()

print(median_age_by_class)

Pclass
1    37.0
2    29.0
3    24.0
Name: Age, dtype: float64


In [27]:
train_dataset_update.head(50)

PassengerId  Survived  Pclass  \
0             1         0       3   
1             2         1       1   
2             3         1       3   
3             4         1       1   
4             5         0       3   
5             6         0       3   
6             7         0       1   
7             8         0       3   
8             9         1       3   
9            10         1       2   
10           11         1       3   
11           12         1       1   
12           13         0       3   
13           14         0       3   
14           15         0       3   
15           16         1       2   
16           17         0       3   
17           18         1       2   
18           19         0       3   
19           20         1       3   
20           21         0       2   
21           22         1       2   
22           23         1       3   
23           24         1       1   
24           25         0       3   
25           26         1       3   
26           27         0       3   
27           28         0       1   
28           29         1       3   
29           30         0       3   
30           31         0       1   
31           32         1       1   
32           33         1       3   
33           34         0       2   
34           35         0       1   
35           36         0       1   
36           37         1       3   
37           38         0       3   
38           39         0       3   
39           40         1       3   
40           41         0       3   
41           42         0       2   
42           43         0       3   
43           44         1       2   
44           45         1       3   
45           46         0       3   
46           47         0       3   
47           48         1       3   
48           49         0       3   
49           50         0       3   

                                                 Name  Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    1  22.0      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1   
2                              Heikkinen, Miss. Laina    0  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1   
4                            Allen, Mr. William Henry    1  35.0      0   
5                                    Moran, Mr. James    1   NaN      0   
6                             McCarthy, Mr. Timothy J    1  54.0      0   
7                      Palsson, Master. Gosta Leonard    1   2.0      3   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)    0  27.0      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)    0  14.0      1   
10                    Sandstrom, Miss. Marguerite Rut    0   4.0      1   
11                           Bonnell, Miss. Elizabeth    0  58.0      0   
12                     Saundercock, Mr. William Henry    1  20.0      0   
13                        Andersson, Mr. Anders Johan    1  39.0      1   
14               Vestrom, Miss. Hulda Amanda Adolfina    0  14.0      0   
15                   Hewlett, Mrs. (Mary D Kingcome)     0  55.0      0   
16                               Rice, Master. Eugene    1   2.0      4   
17                       Williams, Mr. Charles Eugene    1   NaN      0   
18  Vander Planke, Mrs. Julius (Emelia Maria Vande...    0  31.0      1   
19                            Masselmani, Mrs. Fatima    0   NaN      0   
20                               Fynney, Mr. Joseph J    1  35.0      0   
21                              Beesley, Mr. Lawrence    1  34.0      0   
22                        McGowan, Miss. Anna "Annie"    0  15.0      0   
23                       Sloper, Mr. William Thompson    1  28.0      0   
24                      Palsson, Miss. Torborg Danira    0   8.0      3   
25  Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...    0  38.0      1   
26                            Emir, Mr. Farred Chehab    1   NaN      0   
27              

In [28]:
# Initialize LabelEncoder
label_encoder = LabelEncoder()

# Apply Label Encoding to categorical columns 'Sex' and 'Embarked' separately
for col in ['Sex', 'Embarked']:
    train_dataset_update1[col] = label_encoder.fit_transform(train_dataset_update1[col])

#Optionally, If you need to transform based on train_dataset_update:
#for col in ['Sex', 'Embarked']:
#    train_dataset_update1[col] = label_encoder.fit_transform(train_dataset_update[col])